src/components/views/rooms/MessageComposer.js

```diff
@@ -61,6 +64,7 @@ export default class MessageComposer extends React.Component {
             },
             showFormatting: SettingsStore.getValue('MessageComposer.showFormatting'),
             isQuoting: Boolean(RoomViewStore.getQuotingEvent()),
+            tombstone: this._getRoomTombstone(),
```

Comment:
Could we put this in the parent of the composer (`RoomView` or a new in-between component)? Apart from where it appears on the screen, it's not really related to composing messages.

Reply:
With "this", I mean the whole tombstone code and show-a-message-instead-of-composer logic

Reply:
True, although actually MessageComposer is kind of already a (probably badly named) wrapper around MessageComposerInput which is the actual text field. We'd also have to split out the avatar & room encryption icon as we still want these displayed.

Reply:
Hmm, ok, that does sound painful. Better leave it as it is then.